cypress/e2e/ui/Settings/Application-Settings/schedule.cy.js

```diff
@@ -0,0 +1,306 @@
+    cy.get('input#request').should('exist');
+
+    // Checking for Object
+    cy.get('h3[name="object_attributes"]').should('exist');
+    // Checking for Type Combobox
+    cy.get('label[for="target_class"]').should('exist');
+    cy.get('input[role="combobox"]#target_class').should('exist');
+    // Checking for Object Combobox
+    cy.get('label[for="target_id"]').should('exist');
+    cy.get('input[role="combobox"]#target_id').should('exist');
+
+    // Checking for Attribute/Value pairs
+    cy.contains('h3', 'Attribute/Value Pairs').should('exist');
+    // Checking for 5 attribute-value pairs text fields
+    cy.get('input#attribute_1').should('exist');
+    cy.get('input#value_1').should('exist');
+    cy.get('input#attribute_2').should('exist');
+    cy.get('input#value_2').should('exist');
+    cy.get('input#attribute_3').should('exist');
+    cy.get('input#value_3').should('exist');
+    cy.get('input#attribute_4').should('exist');
+    cy.get('input#value_4').should('exist');
+    cy.get('input#attribute_5').should('exist');
+    cy.get('input#value_5').should('exist');
+
+    /* ===== Selecting "Once" option from "Run" dropdown does not show the "Every" dropdown ===== */
+
+    cy.get('select#timer_typ').select('Once');
+    // Checking whether the Every dropdown is hidden
+    cy.get('input#timer_value').should('not.exist');
+
+    /* ===== Selecting any other option other than "Once" from "Run" dropdown shows the "Every" dropdown ===== */
+
+    cy.get('select#timer_typ').select('Hours', { force: true });
+    // Checking whether the "Every" dropdown exist
+    cy.get('label[for="timer_value"]').should('exist');
+    cy.get('select#timer_value').should('exist');
+
+    cy.get('select#timer_typ').select('Days', { force: true });
+    // Checking whether the "Every" dropdown exist
+    cy.get('label[for="timer_value"]').should('exist');
+    cy.get('select#timer_value').should('exist');
+
+    cy.get('select#timer_typ').select('Weeks', { force: true });
+    // Checking whether the "Every" dropdown exist
+    cy.get('label[for="timer_value"]').should('exist');
+    cy.get('select#timer_value').should('exist');
+
+    cy.get('select#timer_typ').select('Months', { force: true });
+    // Checking whether the "Every" dropdown exist
+    cy.get('label[for="timer_value"]').should('exist');
+    cy.get('select#timer_value').should('exist');
+  });
+
+  it('Checking whether Cancel button works on the Add form', () => {
+    selectConfigMenu();
+    cy.contains('#main-content .bx--btn-set button[type="button"]', 'Cancel')
+      .should('be.enabled')
+      .click();
+    cy.get('#main_div #flash_msg_div .alert-success').contains(
+      'Add was cancelled by the user'
+    );
+  });
+
+  it('Checking whether add, edit & delete schedule works', () => {
+    /* ===== Adding a schedule ===== */
+    addSchedule();
+    cy.get('#main_div #flash_msg_div .alert-success').contains(
+      'Schedule "Test name" was saved'
```

Comment:
Note, we'll need to be careful of side effect test which  modify the tables that are presented in the UI as a failure could fail to clean up after itself.   This could cause other tests to fail, for example, if they are listing schedules and asserting the contents.

Reply:
Right, my comment was more about cleaning up afterwards.  So, keep side effects, such as new records that are displayed in the UI, as minimal as possible and we can eventually move to doing cypress-rails way with transactions that rollback changes by labeling side effect tests and doing this rollback for these.